### PR TITLE
Doc: development practices: move relevant content from RFC8

### PR DESCRIPTION
**Pull request by rouault**
Supersedes #9689

### CODE REVIEW

File: doc/source/development/dev_practices.rst [hunk 1]
```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
...
+        ints with only TRUE/FALSE values.
+-  *by*: byte (GByte / unsigned char).
+-  *df*: floating point value (double precision)
+-  *e*: enumeration
+-  *i*: integer number used as a zero based array or loop index.
+-  *f*: floating point value (single precision)
+-  *h*: an opaque handle (such as GDALDatasetH).
+-  *n*: integer number (size unspecified)
+-  *o*: C++ object
+-  *os*: CPLString
```
Comment (dbaston) [inline doc/source/development/dev_practices.rst:62]:
```suggestion
-  *os*: CPLString or std::string
```

File: doc/source/development/dev_practices.rst [hunk 2]
```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
...
+-  *p*: pointer
+-  *psz*: pointer to a zero terminated string. (eg. "char \*pszName;")
+-  *sz*: zero terminated string (eg." char szName[100];")
```
Comment (dbaston) [inline doc/source/development/dev_practices.rst:65]:
```suggestion
-  *sz*: null-terminated string (eg." char szName[100];")
```

File: doc/source/development/dev_practices.rst [hunk 3]
```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
...
+-  *df*: floating point value (double precision)
+-  *e*: enumeration
+-  *i*: integer number used as a zero based array or loop index.
+-  *f*: floating point value (single precision)
+-  *h*: an opaque handle (such as GDALDatasetH).
+-  *n*: integer number (size unspecified)
+-  *o*: C++ object
+-  *os*: CPLString
+-  *p*: pointer
+-  *psz*: pointer to a zero terminated string. (eg. "char \*pszName;")
```
Comment (dbaston) [inline doc/source/development/dev_practices.rst:64]:
```suggestion
-  *psz*: pointer to a null-terminated string. (eg. "char \*pszName;")
```

File: doc/source/development/dev_practices.rst [hunk 4]
```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
...
+-  *sz*: zero terminated string (eg." char szName[100];")
+-  *k*: compile-time constant
+
+Prefix can be stacked. The following are some examples of meaningful
+variables.
+
+-  \*char !\*\ *papszTokens*: Pointer to the an array of strings.
```
Comment (dbaston) [inline doc/source/development/dev_practices.rst:71]:
```suggestion
-  \*char !\*\ *papszTokens*: Pointer to an array of strings.
```

File: doc/source/development/dev_practices.rst [hunk 5]
```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
...
+-  *h*: an opaque handle (such as GDALDatasetH).
+-  *n*: integer number (size unspecified)
+-  *o*: C++ object
+-  *os*: CPLString
+-  *p*: pointer
+-  *psz*: pointer to a zero terminated string. (eg. "char \*pszName;")
+-  *sz*: zero terminated string (eg." char szName[100];")
+-  *k*: compile-time constant
+
+Prefix can be stacked. The following are some examples of meaningful
```
Comment (dbaston) [inline doc/source/development/dev_practices.rst:68]:
```suggestion
Prefixes can be stacked. The following are some examples of meaningful
```

File: doc/source/development/dev_practices.rst (hunk 2 above)
Comment (rouault) [reply doc/source/development/dev_practices.rst:65]: should that be NUL-terminated string ? I believe NUL is the idimomatic way to designate character of code 0

Comment (rouault) [reply doc/source/development/dev_practices.rst:65]: https://en.wikipedia.org/wiki/Null-terminated_string confirms your proposal. and also that a "null-terminate" string is terminated by character NUL. :-)

**Pull request merged**